Alignment/OfflineValidation/bin/GCPtrends.cc

```diff
@@ -0,0 +1,272 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include "TROOT.h"
+#include "TFile.h"
+#include "TChain.h"
+#include "TSelector.h"
+#include "TTreeReader.h"
+#include "TTreeReaderValue.h"
+#include "TTreeReaderArray.h"
+#include "TGraphErrors.h"
+#include "TH1D.h"
+#include "TObject.h"
+#include "TMath.h"
+#include "TString.h"
+
+#include "Alignment/OfflineValidation/interface/Trend.h"
+
+int main(int argc, char *argv[]) {
+  if (argc < 2) {
+    std::cout << "Usage of the program : GCPtrends [gcp_trend_config.json]" << std::endl;
+    std::cout << "gcp_trend_config.json : file with configuration in json format" << std::endl;
+    exit(1);
+  }
+
+  // parsing input file
+  std::string lumi_file(argv[1]);
+  boost::property_tree::ptree config_root;
+  boost::property_tree::read_json(lumi_file, config_root);
+
+  // configuration
+  TString outputdir = config_root.get<std::string>("outputdir", ".");
+  outputdir = TString(outputdir) + TString(outputdir.EndsWith("/") ? "" : "/");
+
+  bool usebadmodules = config_root.get<bool>("usebadmodules", true);
+
+  bool splitpixel = config_root.get<bool>("splitpixel", false);
+
+  Int_t trackerphase = config_root.get<int>("trackerphase");
+
+  TString lumitype = config_root.get<std::string>("trends.lumitype");
+
+  Int_t firstrun = config_root.get<int>("trends.firstrun");
+
+  Int_t lastrun = config_root.get<int>("trends.lastrun");
+
+  std::string lumibyrunfile = config_root.get<std::string>("lumibyrunfile", "");
+  std::ifstream lumifile(lumibyrunfile);
+  assert(lumifile.good());
+  const Run2Lumi lumi_per_run(lumibyrunfile, firstrun, lastrun, 1000);
+
+  std::map<TString, Int_t> comparison_map;
+  for (boost::property_tree::ptree::value_type &comparison : config_root.get_child("comparisons")) {
+    TString path = comparison.first;
+    Int_t run_label = comparison.second.get_value<int>();
+
+    comparison_map[path] = run_label;
+  }
+
+  // create output file
+  TFile *file_out = TFile::Open(outputdir + "GCPTrends.root", "RECREATE");
+  file_out->cd();
+
+  gErrorIgnoreLevel = kError;
+
+  // sub-detector mapping
+  std::map<Int_t, TString> Sublevel_Subdetector_Map = {
+      {1, "PXB"}, {2, "PXF"}, {3, "TIB"}, {4, "TID"}, {5, "TOB"}, {6, "TEC"}};
+
+  // wheels/layers per tracker phase: <pahse,<sublevel,number of layers/wheels>>
+  const std::map<Int_t, std::map<Int_t, Int_t>> Phase_Subdetector_Layers_Map = {
+      {0, {{1, 3}, {2, 2}}}, {1, {{1, 4}, {2, 3}}}, {2, {{1, 4}, {2, 10}}}};
```

Comment:
there are twelve Disks in the phase-2  tracker and not 10.

![image](https://user-images.githubusercontent.com/5082376/208082889-47782bad-7c41-410c-96bf-f4c90ed7a2c0.png)

also the access patter of `std::map` of `std::map`s is not really user-friendly... perhaps this could be recast into a class with accessors

Reply:
Thanks a lot @mmusich for the comments, and sorry for the delay (I was offline). I have implemented most of your comments, and my only minor follow up is on this one; I took the numbers from this [table](https://github.com/cms-sw/cmssw/tree/master/Geometry/TrackerGeometryBuilder) on the relevant package on CMSSW, maybe that is outdated then? In any case I trust your plot :)

Reply:
> I took the numbers from this [table](https://github.com/cms-sw/cmssw/tree/master/Geometry/TrackerGeometryBuilder) on the relevant package on CMSSW, maybe that is outdated then?

Indeed, it looks like it was not updated.

Reply:
Anyway you don't have to trust me, construct a `TrackerGeometry` object and see it for yourself ;)
For reference, here's the output of this test in the same package you pointed out:

https://github.com/cms-sw/cmssw/blob/c628e55f1326b4ab00a83c2e01133feba6aeeb7c/Geometry/TrackerGeometryBuilder/test/BuildFile.xml#L37-L41

https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc11/CMSSW_13_0_X_2023-01-16-1100/unitTestLogs/Geometry/TrackerGeometryBuilder#/82020